version: "3.8"
services:
  test_db:
    image: {{ $mysqlImage }}
    container_name: {{ $dockerTagNamespace }}-test-mysql
    restart: unless-stopped
    tty: true
    ports:
      - "{{ $testDbHostPort }}:3306"
    volumes:
      - {{ $dockerTagNamespace }}-test-mysql:/var/lib/mysql/
    networks:
      - {{ $dockerTagNamespace }}-network
    env_file: ../../../.env
    environment:
      MYSQL_DATABASE: ${TEST_DB_DATABASE}
      MYSQL_USER: ${TEST_DB_USERNAME}
      MYSQL_PASSWORD: ${TEST_DB_PASSWORD}
      MYSQL_ROOT_PASSWORD: ${TEST_DB_ROOT_PASSWORD}
    healthcheck: 
      test: ["CMD-SHELL",  "/usr/bin/mysql --user=${TEST_DB_USERNAME} --password=${TEST_DB_PASSWORD} --execute \"SHOW DATABASES;\""]
      interval: 10s
      timeout: 2s
      retries: 10
  mailhog:
    image: 'mailhog/mailhog:latest'
    container_name: {{ $dockerTagNamespace }}-mailhog
    restart: unless-stopped
    ports:
        - "${MAILHOG_PORT:-1025}:1025"
        - "${MAILHOG_DASHBOARD_PORT:-8025}:8025"
    networks:
        - "{{ $dockerTagNamespace }}-network"
volumes:
  {{ $dockerTagNamespace }}-test-mysql:
    driver: local
